source/components/text-input/text-input.less

```diff
@@ -38,7 +42,7 @@ textarea {
     }
   }
 
-  &.-hxInvalid {
+  &[invalid] {
```

Comment:
This does not interfere with [global HTML Attributes](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes).